rules/ast/ast.go

```diff
@@ -371,10 +383,15 @@ func (node *ScalarFunctionCall) Eval(timestamp clientmodel.Timestamp) clientmode
 	return node.function.callFn(timestamp, node.args).(clientmodel.SampleValue)
 }
 
-func (node *VectorAggregation) labelsToGroupingKey(labels clientmodel.Metric) uint64 {
+// hashForLabels returns a hash value taken from the values of the
+// specified labels in the metric. The hash is not based on the label names.
+func hashForLabels(metric clientmodel.Metric, labels clientmodel.LabelNames) uint64 {
+	// Sort labels to guarantee the same hash across numerous calls.
```

Comment:
True, you'd have to always include the label names in each subhash: `<name> <separator> <value>`.

Not sure what's more expensive: hashing ~double the amount of strings, or doing the pre-sort. Probably doesn't matter much. Fine as it is for now then I guess :)